bgremoval_package/demo/run.py

```diff
@@ -0,0 +1,208 @@
+from pathlib import Path
+from IP_LAP.bgremoval_package.src.models.modnet import MODNet
+
+
+#Check if GPU available or not
+GPU = True if torch.cuda.device_count() > 0 else False
+
+
+#Preparing path for pretrained checkpoint
+filepath = os.path.dirname(os.path.abspath(__file__))
+CKPT_FOLDER = Path(filepath).parent.parent.parent
+PRETRAINED_CKPT = os.path.join(CKPT_FOLDER,"checkpoints/bgremoval/modnet_webcam_portrait_matting.ckpt")
+print(PRETRAINED_CKPT)
+torch_transforms = transforms.Compose(
+    [
+        transforms.ToTensor(),
+        transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5)),
+    ]
+)
+def load_model():
+    
+    modnet = MODNet(backbone_pretrained=False)
+    modnet = nn.DataParallel(modnet)
+
+    
+    if GPU:
+        print('Use GPU...')
+        modnet = modnet.cuda()
+        modnet.load_state_dict(torch.load(PRETRAINED_CKPT))
+    else:
+        print('Use CPU...')
+        modnet.load_state_dict(torch.load(PRETRAINED_CKPT, map_location=torch.device('cpu')))
+    modnet.eval()
+    return modnet
+
+def matting_list(video,output_folder):
+    
+    modnet = load_model()
+    if not os.path.exists(output_folder):
+        os.makedirs(output_folder)
+    frame=video[0]
+
+    num_frame = len(video)
+    h, w = frame.shape[:2]
+    if w >= h:
+        rh = 512
+        rw = int(w / h * 512)
+    else:
+        rw = 512
+        rh = int(h / w * 512)
+    rh = rh - rh % 32
+    rw = rw - rw % 32
+
+
+    print('Start matting...')
+    for c in tqdm(range(int(num_frame))):
+        frame=video[c]
+        # frame_np = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
+        frame_np = cv2.resize(frame, (rw, rh), cv2.INTER_AREA)
+
+        frame_PIL = Image.fromarray(frame_np)
+        frame_tensor = torch_transforms(frame_PIL)
+        frame_tensor = frame_tensor[None, :, :, :]
+        GPU = True if torch.cuda.device_count() > 0 else False
+        if GPU:
+            frame_tensor = frame_tensor.cuda()
+
+        with torch.no_grad():
+            _, _, matte_tensor = modnet(frame_tensor, True)
+
+        matte_tensor = matte_tensor.repeat(1, 3, 1, 1)
+        matte_np = matte_tensor[0].data.cpu().numpy().transpose(1, 2, 0)
+        view_np_alpha = matte_np * np.full(frame_np.shape, 255.0)
+        view_np_alpha = cv2.cvtColor(view_np_alpha.astype(np.uint8), cv2.COLOR_RGB2BGR)
+        view_np_alpha = cv2.resize(view_np_alpha, (w, h))
+
+
+        view_np = matte_np * frame_np + (1 - matte_np) * np.full(frame_np.shape, 255.0)
+        view_np = cv2.cvtColor(view_np.astype(np.uint8), cv2.COLOR_RGB2BGR)
+        view_np = cv2.resize(view_np, (w, h))
+        view_np_alpha=view_np_alpha[:,:,1]
+        four_channel_frame = np.dstack((view_np, view_np_alpha))
+        output_path = os.path.join(output_folder, f'output_{c:04d}.png')
+        cv2.imwrite(output_path, four_channel_frame)
+
+
+def matting(video,output_folder, alpha_matte=False, fps=25):
+    # video capture
+    modnet = load_model()
```

Comment:
Initialize outside this function when class is initialize.

Reply:
ok..will do it...